paddle/fluid/framework/executor.cc

```diff
@@ -466,7 +467,7 @@ void Executor::RunPreparedContext(ExecutorPrepareContext* ctx, Scope* scope,
       // the sub scopes it created should not be dropped immediately, because
       // while_grad_op will use some variables created during while_op run, so
       // we need to keep the kids and wait for the outer executor to drop them.
-      scope->DropKids();
+      Async([scope] { scope->DropKids(); });
```

Comment:
what if the thread is not started before the next Run starts?

Reply:
I think it's fine. Because each time the executor will create a brand new kid scope. 
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/executor.cc#L420